wfe/wfe.go

```diff
@@ -1546,40 +1496,34 @@ func (wfe *WebFrontEndImpl) KeyRollover(ctx context.Context, logEvent *requestEv
 	}
 	err = json.Unmarshal(payload, &rolloverRequest)
 	if err != nil {
-		logEvent.AddError("unable to JSON parse resource from JWS payload: %s", err)
 		wfe.sendError(response, logEvent, probs.Malformed("Request payload did not parse as JSON"), nil)
 		return
 	}
 
 	if wfe.relativeEndpoint(request, fmt.Sprintf("%s%d", regPath, reg.ID)) != rolloverRequest.Account {
-		logEvent.AddError("incorrect account URL provided")
 		wfe.sendError(response, logEvent, probs.Malformed("Incorrect account URL provided in payload"), nil)
 		return
 	}
 
 	keysEqual, err := core.PublicKeysEqual(rolloverRequest.NewKey.Key, newKey.Key)
 	if err != nil {
-		logEvent.AddError("unable to marshal new key: %s", err)
 		wfe.sendError(response, logEvent, probs.Malformed("Unable to marshal new JWK"), nil)
```

Comment:
Maybe `err` should be given to `sendError` here instead of `nil`?

Reply:
This is a weird thing about `sendError` - it only sends the "internal" error when the "external" problem type is ServerInternal. I'd like to clean this up (e.g. split into two functions, one of which doesn't take an "internal" error) as a followup change. However, I can see an argument that this is an example where it is indeed useful to leave the `AddError` in place.